### PR TITLE
[MDB IGNORE] Introducing old pipes

### PR DESCRIPTION
**Pull request by TheChosenEvilOne**
just opening to keep track.
- [x] sed all maps
- [x] unwrenching broke, fix it

### CODE REVIEW

**Comment by JinGiant** [comment]
Is this like smart pipes but they're actually smart and not retarded

**Comment by TheChosenEvilOne** [comment]
> Is this like smart pipes but they're actually smart and not retarded

this just reverts smart pipes, and adds a non-retarded mapping helper.

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -391,6 +391,77 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+			else if (straight) // We don't really care about components in straight mode
+				continue
+			p_dir |= d
+	p_dir &= ~taken_dirs
+	if (straight && !p_dir)
+		p_dir = taken_dirs & 3 ? EAST : NORTH
+	if (p_dir == 0xF)
+		spawn_pipe(/obj/machinery/atmospherics/pipe/manifold4w)
+	//in binary: (p_dir & 0b0011 > 0b0001 && p_dir & 0b1100) || (p_dir & 0b0011 && p_dir & 0b1100 > 0b0100)
+	else if (((p_dir & 0x3) == 0x3 && p_dir & 0xC) || (p_dir & 0x3 && (p_dir & 0xC) == 0xC))
```
Comment (TheChosenEvilOne) [inline code/modules/mapping/mapping_helpers.dm:447]:
```suggestion
	else if ((p_dir >> 2 + p_dir & 0x3) > 3)
```
should work

File: code/modules/mapping/mapping_helpers.dm [hunk 2]
```diff
@@ -391,6 +391,77 @@ INITIALIZE_IMMEDIATE(/obj/effect/mapping_helpers/no_lava)
...
+		spawn_pipe(/obj/machinery/atmospherics/pipe/manifold, p_dir ^ 0xF)
+	else if (p_dir)
+		if ((p_dir & 0x3) == 0x3 || (p_dir & 0xC) == 0xC)
```
Comment (TheChosenEvilOne) [inline code/modules/mapping/mapping_helpers.dm:450]:
```suggestion
		if ((p_dir - 1 & p_dir) & 0x2)
```
should work

**Comment by karmaisblackandbluepilled** [comment]
@TheChosenEvilOne WAKE UP
WAKE UP HOW THE FUCK DO WE CONTACT YOU


ALSO BY THE WAY YOU PORTED SEPARATED CHEMS AND THEY DON'T WORK, FIX THEM

**Pull request merged**